crates/plugins/src/manager.rs

```diff
@@ -60,7 +61,9 @@ impl PluginManager {
         let target_url = Url::parse(&target)?;
         let temp_dir = tempdir()?;
         let plugin_tarball_path = match target_url.scheme() {
-            URL_FILE_SCHEME => PathBuf::from(target_url.path()),
+            URL_FILE_SCHEME => target_url
+                .to_file_path()
+                .map_err(|_| anyhow!("Invalid file URL: {target_url:?}"))?,
```

Comment:
I _almost_ just suggested `.context` here, which goes to show how confusing `to_file_path` is :sweat_smile: